File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/li_std_vector_enum_runme.rb
#!/usr/bin/env ruby

require 'swig_assert'
require 'li_std_vector_enum'
include Li_std_vector_enum

ev = EnumVector.new()

swig_assert(ev.nums[0] == 10)
swig_assert(ev.nums[1] == 20)
swig_assert(ev.nums[2] == 30)

it = ev.nums.begin
v = it.value()
swig_assert(v == 10)
it.next()
v = it.value()
swig_assert(v == 20)

expected = 10 
ev.nums.each do|val|
  swig_assert(val == expected)
  expected += 10
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/li_std_vector_enum_runme.py
import li_std_vector_enum


def check(a, b):
    if (a != b):
        raise RuntimeError("Not equal: ", a, b)

ev = li_std_vector_enum.EnumVector()

check(ev.nums[0], 10)
check(ev.nums[1], 20)
check(ev.nums[2], 30)

it = ev.nums.iterator()
v = it.value()
check(v, 10)
it.next()
v = it.value()
check(v, 20)

expected = 10
for val in ev.nums:
    check(val, expected)
    expected += 10
